com.unity.ml-agents/Runtime/Inference/ApplierImpl.cs

```diff
@@ -165,10 +170,6 @@ internal void ComputeCdf(TensorProxy logProbs, int batch, int channelOffset, int
         {
             // Find the class maximum
             var maxProb = float.NegativeInfinity;
-            if (logProbs.Device == DeviceType.GPU)
-            {
-                logProbs.data.MakeReadable();
```

Comment:
Should you add a CompleteOperationsAndDownload() in this method?

Reply:
Yes. Good catch. This is for legacy support, but yeah, should be CompleteOperationsAndDownload for non CPU.